src/wazuh_modules/vulnerability_scanner/src/scanOrchestrator/versionMatcher/versionMatcher.hpp

```diff
@@ -142,6 +153,11 @@ class VersionMatcher final
         auto pVersionObjectA = createVersionObject(versionA, type);
         auto pVersionObjectB = createVersionObject(versionB, type);
 
+        if (!pVersionObjectA || !pVersionObjectB)
+        {
+            return INVALID_VERSION_OBJECT_TYPE;
+        }
+
         if (pVersionObjectA->getType() == pVersionObjectB->getType())
```

Comment:
Maybe this conditional could be modified to absorb the previous one.
`if(pVersionObjectA && pVersionObjectB && pVersionObjectA->getType() == pVersionObjectB->getType())`

Reply:
done